subprojects/docs/src/docs/release/notes.md

```diff
@@ -78,6 +78,48 @@ The `buildSrc` build of the root project continues to use its own build cache co
 However, the `buildSrc` build of any included build will inherit the build cache configuration from the root build.
 For more on configuring the build cache for the root `buildSrc` build, please see [the Userguide section on this topic](userguide/build_cache.html#buildCacheBuildSrc).
 
+### Richer task inputs
```

Comment:
I don't like "richer" here. "Rich task inputs" seems marginally better. "Polymorphic task properties" covers the technicals best IMO, but it's also a mouthful I'd rather replace with simpler words. What we ultimately want tell users about is the new use of dynamic types when declaring nested task inputs and outputs. As it is hard to explain the concept, it would be nice to demonstrate with an example instead.

We should also link to the user manual from here.

Reply:
I reworked the release notes and added a section to the userguide. Please have another look.